packages/cli-hydrogen/src/cli/services/deploy/error.ts

```diff
@@ -0,0 +1,24 @@
+import {system, error} from '@shopify/cli-kit'
+
+export const WebPageNotAvailable = () => {
+  return new error.Abort('Web page is not available.')
+}
+
+export const retryOnError = async <T>(
+  execute: () => T,
+  maxIteration = 5,
+  waitTime: number | ((n: number) => number) = 1,
+) => {
+  for (const iteration of [...Array(maxIteration)]) {
+    try {
+      // eslint-disable-next-line no-await-in-loop
+      return await execute()
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (error: any) {
+      // eslint-disable-next-line no-await-in-loop
+      await system.sleep(typeof waitTime === 'number' ? waitTime : waitTime(iteration))
+
+      if (iteration + 1 === maxIteration) throw new error.Abort(error.message)
+    }
+  }
+}
```

Comment:
@tempor1s Created a new file called `deploy/error.ts` where we can put all our error related stuff there. Made sense to me for the work still ahead.